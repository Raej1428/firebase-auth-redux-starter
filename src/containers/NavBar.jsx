// @flow
import { connect } from 'react-redux';
import { signOutUser } from '../actions/auth';
import NavBar from '../components/NavBar';

const mapStateToProps = state => ({
  isAuthenticated: state.auth.authenticated,
});

const mapDispatchToProps = dispatch => ({
  signOut: () => dispatch(signOutUser()),
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
