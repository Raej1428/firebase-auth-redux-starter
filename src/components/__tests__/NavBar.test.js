/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import NavBar, { NavigationAuth, NavigationNonAuth } from '../NavBar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('NavBar component', () => {
  it('matches snapshot for non-authed', () => {
    const wrapper = render(
      <Router>
        <NavBar isAuthenticated={false} />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('matches snapshot for authed', () => {
    const wrapper = render(
      <Router>
        <NavBar isAuthenticated={false} />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  })

})
