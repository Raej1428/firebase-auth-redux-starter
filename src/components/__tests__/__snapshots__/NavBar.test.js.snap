// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot for authed 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "flex justify-between bb bg-black b--white-10",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "href": "/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "link white-70 hover-white no-underline flex items-center pa3",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "dib h1 w1",
                      "data-icon": "grid",
                      "style": "fill:currentcolor",
                      "viewBox": "0 0 32 32",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Super Normal Icon Mark",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "title",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": Object {
                          "attribs": Object {
                            "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                      Object {
                        "attribs": Object {
                          "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Super Normal Icon Mark",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "title",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-icon": undefined,
                      "style": undefined,
                      "viewBox": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-icon": undefined,
                      "style": undefined,
                      "viewBox": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "flex-grow pa3 flex items-center",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/signin",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "f6 dib white bg-animate hover-bg-white hover-black no-underline pv2 ph4 br-pill ba b--white-20",
                        "href": "#0",
                      },
                      "children": Array [
                        Object {
                          "data": "Login",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "href": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "flex-grow pa3 flex items-center",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/signin",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "f6 dib white bg-animate hover-bg-white hover-black no-underline pv2 ph4 br-pill ba b--white-20",
                      "href": "#0",
                    },
                    "children": Array [
                      Object {
                        "data": "Login",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "href": "/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "link white-70 hover-white no-underline flex items-center pa3",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "dib h1 w1",
                        "data-icon": "grid",
                        "style": "fill:currentcolor",
                        "viewBox": "0 0 32 32",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Super Normal Icon Mark",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "title",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": Object {
                            "attribs": Object {
                              "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        Object {
                          "attribs": Object {
                            "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "Super Normal Icon Mark",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "title",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "data-icon": undefined,
                        "style": undefined,
                        "viewBox": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "data-icon": undefined,
                        "style": undefined,
                        "viewBox": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "href": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "nav",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`NavBar component matches snapshot for non-authed 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "flex justify-between bb bg-black b--white-10",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "href": "/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "link white-70 hover-white no-underline flex items-center pa3",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "dib h1 w1",
                      "data-icon": "grid",
                      "style": "fill:currentcolor",
                      "viewBox": "0 0 32 32",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "Super Normal Icon Mark",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "title",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": Object {
                          "attribs": Object {
                            "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                      Object {
                        "attribs": Object {
                          "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Super Normal Icon Mark",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "title",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-icon": undefined,
                      "style": undefined,
                      "viewBox": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-icon": undefined,
                      "style": undefined,
                      "viewBox": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "flex-grow pa3 flex items-center",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "href": "/signin",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "f6 dib white bg-animate hover-bg-white hover-black no-underline pv2 ph4 br-pill ba b--white-20",
                        "href": "#0",
                      },
                      "children": Array [
                        Object {
                          "data": "Login",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "href": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "href": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "flex-grow pa3 flex items-center",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "href": "/signin",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "f6 dib white bg-animate hover-bg-white hover-black no-underline pv2 ph4 br-pill ba b--white-20",
                      "href": "#0",
                    },
                    "children": Array [
                      Object {
                        "data": "Login",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "href": "/",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "link white-70 hover-white no-underline flex items-center pa3",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "dib h1 w1",
                        "data-icon": "grid",
                        "style": "fill:currentcolor",
                        "viewBox": "0 0 32 32",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "Super Normal Icon Mark",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "title",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": Object {
                            "attribs": Object {
                              "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        Object {
                          "attribs": Object {
                            "d": "M2 2 L10 2 L10 10 L2 10z M12 2 L20 2 L20 10 L12 10z M22 2 L30 2 L30 10 L22 10z M2 12 L10 12 L10 20 L2 20z M12 12 L20 12 L20 20 L12 20z M22 12 L30 12 L30 20 L22 20z M2 22 L10 22 L10 30 L2 30z M12 22 L20 22 L20 30 L12 30z M22 22 L30 22 L30 30 L22 30z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "Super Normal Icon Mark",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "title",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "data-icon": undefined,
                        "style": undefined,
                        "viewBox": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "data-icon": undefined,
                        "style": undefined,
                        "viewBox": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "href": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "nav",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
